avo-hq/avo
# chore: autoreload lib directory and initializer in development

# Description


Fixes an old issue we had in development. When you used to make a change on a base file (eg: `BaseResource`), that change was not visible until you restarted the server. Now it auto-reloads the `lib` directory when doing development on Avo (when `Avo::IN_DEVELOPMENT` not `Rails.env.development?`).

It will also reloads changes in the `avo.rb` initializer for `Rails.env.development?`.

# Checklist:



## Manual review steps


1. Start the app and output `Avo.configuration.license` somewhere on the screen and have the license to `community` in your initializer.
2. Change your initializer to `pro`
3. you should see the change on screen without restarting the server.

Manual reviewer: please leave a comment with output from the test if that's the case.
